include/nil/blueprint/parser.hpp

```diff
@@ -1366,8 +1366,8 @@ namespace nil {
 
             template<typename InputType>
             var put_into_assignment(InputType input) {
-                assignments[currProverIdx].public_input(0, public_input_idx) = input;
-                return var(0, public_input_idx++, false, var::column_type::public_input);
+                assignments[currProverIdx].constant(1, constant_idx) = input;
```

Comment:
Need to add shared variable in case of `next_prover`. OK for me as hot solution, we can do it later.